File: projects/10/ruby/Kiwi.rb
# Kiwi.rb | Defines the Kiwi class which inherits attributes and methods
#   from the Bird superclass.
#
# Begun by: Dr. Adams, for CS 214 at Calvin College.
# Completed by: Sinai Park
# Date: APril 24 2020
####################################################

require './WalkingBird.rb'

class Kiwi < WalkingBird

######################################################
# call returns the sound that a Kiwi type makes       #
# Recieve: NONE                                      #
# Return: Kiii-ki-ki!, a String                           #
######################################################
def call
	"Kiii-ki-ki!"
end

end

File: projects/10/java/Kiwi.java
/* Kiwi.java provides Kiwi subclass of Bird.
 *
 * Begun by: Prof. Adams, for CS 214 at Calvin College.
 * Completed by: Sinai Park
 * Date: April 24 2020
 ******************************************************/

public class Kiwi extends WalkingBird {
   
     /* explicit constructor
      * Receive: name, a String 
      * PostCond: myName == name.
      */
    public Kiwi(String name)
    {
        super(name);
    }
    
    
     /* A Kiwi's Call
      * Return: a Kiwi-call ("Ki-ki-kikiii!").
      */
    public String call()
    {
        return "Ki-ki-kikiii!";
    }
    
    }
    
    